poly/src/domain/utils.rs

```diff
@@ -53,12 +53,13 @@ pub(crate) fn compute_powers<F: Field>(size: usize, g: F) -> Vec<F> {
         .into_par_iter()
         .flat_map(|i| {
             let offset = g.pow(&[(i * num_elem_per_thread) as u64]);
-            let res = compute_powers_and_mul_by_const_serial(num_elem_per_thread, g, offset);
-            assert_eq!(res.len(), num_elem_per_thread);
+            // Compute the size that this chunks' output should be
+            // (num_elem_per_thread, unless there are less than num_elem_per_thread elements remaining)
+            let num_elements_to_compute = min(size - i * num_elem_per_thread, num_elem_per_thread);
```

Comment:
This handles the bug when num_cpus doesn't divide the size?

Reply:
Yeah

Reply:
awesome